### DISCUSSION TITLE
Pass data as prop in csp mode

### DISCUSSION THREAD

**Discussion by Sulaymon-dev** [alpinejs/alpine#1678] [1. Help] [answered]
How i can pass prop to component from outside in csp mode...
Like

```html
<div x-data="counter" :defaultCount="10">
    <button @click="increment">Increment</button>
    <span x-text="count"></span>
</div>
<html>

**Comment by alexej-d** [comment, marked as answer]
Version 3.2 of Alpine allows this form of instantiation:
```
<div x-data="counter({ defaultCount: 10 })">
    <button @click="increment">Increment</button>
    <span x-text="count"></span>
</div>
```

**Reply by ryangjchandler** [reply]
@alexej-d This won't work in CSP mode since it still relies on evaluating the string. For CSP mode, I'd generally recommend using data attributes and sending the values through `JSON.parse`. This will only allow you to use valid JSON but in 99% of the scenarios, that's okay.

**Reply by alexej-d** [reply]
Ah sorry, you are right!
